reconcile/blackbox_exporter_endpoint_monitoring.py

```diff
@@ -159,6 +167,24 @@ def fill_desired_state(provider: EndpointMonitoringProvider,
 @defer
 def run(dry_run: bool, thread_pool_size: int, internal: bool,
         use_jump_host: bool, defer=None) -> None:
+    # verify blackbox-exporter modules
+    settings = queries.get_app_interface_settings()
+    allowed_modules = settings["endpointMonitoringBlackboxExporterModules"]
+    verification_errors = False
+    if allowed_modules:
+        for p in get_blackbox_providers():
+            if p.blackboxExporter and \
+                 p.blackboxExporter.module not in allowed_modules:
```

Comment:
How long is `allowed_modules` in a typical case? This loop will be `O(len(providers)*len(allowed_modules))`

Reply:
i can make allowed_modules a set, i guess that will make it `O(len(providers)*some_constant)`